transitions/registration.js

```diff
@@ -150,57 +170,36 @@ module.exports = {
     onMatch: function(change, db, audit, callback) {
         var self = module.exports,
             doc = change.doc,
-            config = self.getRegistrationConfig(self.getConfig(), doc.form);
+            registrationConfig = self.getRegistrationConfig(self.getConfig(), doc.form);
 
-        if (!config) {
+        if (!registrationConfig) {
             return callback();
         }
 
-        self.validate(config, doc, function(errors) {
-
+        self.validate(registrationConfig, doc, function(errors) {
             if (errors && errors.length > 0) {
-                messages.addErrors(doc, errors);
-                // join all errors into one response or respond with first
-                // error.
-                if (config.validations.join_responses) {
-                    var msgs = [];
-                    _.each(errors, function(err) {
-                        if (err.message) {
-                            msgs.push(err.message);
-                        } else if (err) {
-                            msgs.push(err);
-                        }
-                    });
-                    messages.addReply(doc, msgs.join('  '));
-                } else {
-                    var err = _.first(errors);
-                    messages.addReply(doc, err.message || err);
-                }
+                addValidationErrors(registrationConfig, doc, errors, callback);
                 return callback(null, true);
             }
 
             if (doc.fields && doc.fields.patient_id) {
                 // We're attaching this registration to an existing patient, let's
                 // make sure it's valid
-                // NB: if in the future we allow partners to configure the add_patient
-                //     trigger to support passing in their own patient shortcodes
-                //     we're going to have to change this logic:
-                //     https://github.com/medic/medic-webapp/issues/3000
```

Comment:
This is OK, as long as we document that the external ID they pass cannot be mapped to `patient_id` but must be a different name (like `external_id`). Need to document this somewhere.